### DISCUSSION TITLE
Foreign Keys that reference non pk columns

### DISCUSSION THREAD

**Discussion by theelderbeever** [piccolo-orm/piccolo#393] [Q&A] [answered]
Is it possible to create a foreign key on a table that references a column other than the primary key on a table? The documentation seems to imply that you can only pass a table to `ForeignKey` and it picks up the pk. I would like to do something like the following...

```python
class Contract(BaseTable):

    address = Varchar(index=True, required=True, null=False, unique=True)

class Log(BaseTable):

    contract_address = ForeignKey(Contract.address)
```

Is there a way to accomplish this?

**Comment by dantownsend** [comment, marked as answer]
@theelderbeever It's not currently possible with the version on PyPI.

I've created a PR to add this functionality. It seems OK, but I need to test it a bit more before releasing it.

**Reply by theelderbeever** [reply]
That's excellent to hear! Looking forward to it. 

I glanced at the PR and noticed that you have docs noting the uniqueness requirement of the reference column however, I didn't see any code that checks for said constraint on the referenced column unless I missed it. Would that be worth considering or is just delegating to the database as it is now probably better?

**Reply by dantownsend** [reply]
Good point - I'll see if I can add the unique check in code.
